core/base/inc/TStorage.h

```diff
@@ -113,6 +120,20 @@ R__INTENTIONALLY_UNINIT_BEGIN
 R__INTENTIONALLY_UNINIT_END
 }
 
+// Assign the kIsOnHeap bit in 'bits' based on the pattern seen in uniqueID.
+// See Storage::FilledByObjectAlloc for details.
+// This routine is marked as inline with attribute noinline so that it never
+// inlined and thus can be used in a valgrind suppression file to suppress
+// the known/intentional uninitialized memory read but still be a 'quick'
+// function call to avoid losing performance at object creation.
+R__NEVER_INLINE void TStorage::UpdateIsOnHeap(volatile const UInt_t &uniqueID, volatile UInt_t &bits) {
```

Comment:
Is it still cheaper to inline this, even with `noinline`, or how does performance decrease by moving the implementation into the source file? I would hope to see this questions answered in the above comment.

Reply:
it doubles the over-head in some use cases. (up to 20%)